server.ts

```diff
@@ -113,11 +125,58 @@ createConnection(ormOptions)
             // get a user repository to perform operations with user
             const userRepository = getManager().getRepository(User);
         
+            const exisistingUser = await userRepository.findByIds(req.params.id);
+            if (exisistingUser.length === 0) {
+                res.sendStatus(404);
+            }
+            
             userRepository.delete(req.params.id);
             res.end();
             return;
         })
 
+        app.put('/:id', async (req, res) => {
+            // Prepare output in JSON format
+            const user = {
+                id: +req.params.id,
+                email: req.body.email,
+                name: req.body.name,
+                password: req.body.password,
+                profession: req.body.profession
+            };
+            
+            // First read existing users.
+            // get a user repository to perform operations with user
+            const userRepository = getManager().getRepository(User);
+
+            const exisistingUser = await userRepository.findByIds(req.params.id);
+            if (exisistingUser.length === 0) {
+                res.sendStatus(404);
+                return;
+            }
+
+            const users = await userRepository.save(user);
+
+            res.send(users);
+            return;
+        })
+
+        app.patch('/:id', async (req, res) => {
+                        
+            const userRepository = getManager().getRepository(User);
+
+            const exisistingUser = await userRepository.findByIds(req.params.id);
+            if (exisistingUser.length === 0) {
+                res.sendStatus(404);
+                return;
+            }
+            
+            await userRepository.update(req.params.id, req.body);
```

Comment:
Please research another method to return changed properties.
We can use another method instead of `update` in here.

Reply:
I'll replace `update` with `save`.